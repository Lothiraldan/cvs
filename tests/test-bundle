#!/bin/sh

cp "$TESTDIR"/printenv.py .

echo "====== Setting up test"
hg init test
cd test
echo 0 > afile
hg add afile
hg commit -m "0.0" -d "1000000 0"
echo 1 >> afile
hg commit -m "0.1" -d "1000000 0"
echo 2 >> afile
hg commit -m "0.2" -d "1000000 0"
echo 3 >> afile
hg commit -m "0.3" -d "1000000 0"
hg update -C 0
echo 1 >> afile
hg commit -m "1.1" -d "1000000 0"
echo 2 >> afile
hg commit -m "1.2" -d "1000000 0"
echo "a line" > fred
echo 3 >> afile
hg add fred
hg commit -m "1.3" -d "1000000 0"
hg mv afile adifferentfile
hg commit -m "1.3m" -d "1000000 0"
hg update -C 3
hg mv afile anotherfile
hg commit -m "0.3m" -d "1000000 0"
hg verify
cd ..
hg init empty

echo "====== Bundle --all"
hg -R test bundle --all all.hg

echo "====== Bundle test to full.hg"
hg -R test bundle full.hg empty
echo "====== Unbundle full.hg in test"
hg -R test unbundle full.hg
echo "====== Verify empty"
hg -R empty heads
hg -R empty verify

echo "====== Pull full.hg into test (using --cwd)"
hg --cwd test pull ../full.hg
echo "====== Pull full.hg into empty (using --cwd)"
hg --cwd empty pull ../full.hg
echo "====== Rollback empty"
hg -R empty rollback
echo "====== Pull full.hg into empty again (using --cwd)"
hg --cwd empty pull ../full.hg

echo "====== Pull full.hg into test (using -R)"
hg -R test pull full.hg
echo "====== Pull full.hg into empty (using -R)"
hg -R empty pull full.hg
echo "====== Rollback empty"
hg -R empty rollback
echo "====== Pull full.hg into empty again (using -R)"
hg -R empty pull full.hg

echo "====== Log -R full.hg in fresh empty"
rm -r empty
hg init empty
cd empty
hg -R bundle://../full.hg log

echo "====== Pull ../full.hg into empty (with hook)"
echo '[hooks]' >> .hg/hgrc
echo 'changegroup = python ../printenv.py changegroup' >> .hg/hgrc
#doesn't work (yet ?)
#hg -R bundle://../full.hg verify
hg pull bundle://../full.hg
echo "====== Rollback empty"
hg rollback
cd ..
echo "====== Log -R bundle:empty+full.hg"
hg -R bundle:empty+full.hg log --template="{rev} "
echo ""
echo "====== Pull full.hg into empty again (using -R; with hook)"
hg -R empty pull full.hg

echo "====== Create partial clones"
rm -r empty
hg init empty
hg clone -r 3 test partial
hg clone partial partial2
cd partial
echo "====== Log -R full.hg in partial"
hg -R bundle://../full.hg log
echo "====== Incoming full.hg in partial"
hg incoming bundle://../full.hg
echo "====== Outgoing -R full.hg vs partial2 in partial"
hg -R bundle://../full.hg outgoing ../partial2
echo "====== Outgoing -R does-not-exist.hg vs partial2 in partial"
hg -R bundle://../does-not-exist.hg outgoing ../partial2
cd ..

echo "====== Direct clone from bundle (all-history)"
hg clone full.hg full-clone
hg -R full-clone heads
rm -r full-clone

# test for http://mercurial.selenic.com/bts/issue216
echo "====== Unbundle incremental bundles into fresh empty in one go"
rm -r empty
hg init empty
hg -R test bundle --base null -r 0 ../0.hg
hg -R test bundle --base 0    -r 1 ../1.hg
hg -R empty unbundle -u ../0.hg ../1.hg

# test for 540d1059c802
echo "====== test for 540d1059c802"
hg init orig
cd orig
echo foo > foo
hg add foo
hg ci -m 'add foo'

hg clone . ../copy
hg tag foo

cd ../copy
echo >> foo
hg ci -m 'change foo'
hg bundle ../bundle.hg ../orig

cd ../orig
hg incoming ../bundle.hg
cd ..

# test for http://mercurial.selenic.com/bts/issue1144
echo "===== test that verify bundle does not traceback"
# partial history bundle, fails w/ unkown parent
hg -R bundle.hg verify
# full history bundle, refuses to verify non-local repo
hg -R all.hg verify
# but, regular verify must continue to work
hg -R orig verify

echo "====== diff against bundle"
hg init b
cd b
hg -R ../all.hg diff -r tip
cd ..


