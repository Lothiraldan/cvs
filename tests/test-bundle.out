====== Setting up test
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
1 files updated, 0 files merged, 2 files removed, 0 files unresolved
checking changesets
checking manifests
crosschecking files in changesets and manifests
checking files
4 files, 9 changesets, 7 total revisions
====== Bundle test to full.hg
searching for changes
9 changesets found
====== Unbundle full.hg in test
adding changesets
adding manifests
adding file changes
added 0 changesets with 0 changes to 4 files
(run 'hg update' to get a working copy)
====== Verify empty
changeset:   -1:000000000000
tag:         tip
user:        
date:        Thu Jan 01 00:00:00 1970 +0000

checking changesets
checking manifests
crosschecking files in changesets and manifests
checking files
0 files, 0 changesets, 0 total revisions
====== Pull full.hg into test (using --cwd)
pulling from ../full.hg
searching for changes
no changes found
====== Pull full.hg into empty (using --cwd)
pulling from ../full.hg
requesting all changes
adding changesets
adding manifests
adding file changes
added 9 changesets with 7 changes to 4 files (+1 heads)
(run 'hg heads' to see heads, 'hg merge' to merge)
====== Rollback empty
rolling back last transaction
====== Pull full.hg into empty again (using --cwd)
pulling from ../full.hg
requesting all changes
adding changesets
adding manifests
adding file changes
added 9 changesets with 7 changes to 4 files (+1 heads)
(run 'hg heads' to see heads, 'hg merge' to merge)
====== Pull full.hg into test (using -R)
pulling from full.hg
searching for changes
no changes found
====== Pull full.hg into empty (using -R)
pulling from full.hg
searching for changes
no changes found
====== Rollback empty
rolling back last transaction
====== Pull full.hg into empty again (using -R)
pulling from full.hg
requesting all changes
adding changesets
adding manifests
adding file changes
added 9 changesets with 7 changes to 4 files (+1 heads)
(run 'hg heads' to see heads, 'hg merge' to merge)
====== Log -R full.hg in fresh empty
changeset:   8:836ac62537ab
tag:         tip
parent:      3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3m

changeset:   7:80fe151401c2
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3m

changeset:   6:1e3f6b843bd6
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3

changeset:   5:024e4e7df376
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.2

changeset:   4:5f4f3ceb285e
parent:      0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.1

changeset:   3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3

changeset:   2:d62976ca1e50
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.2

changeset:   1:10b2180f755b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.1

changeset:   0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.0

====== Pull ../full.hg into empty (with hook)
changegroup hook: HG_NODE=5649c9d34dd87d0ecb5fd39672128376e83b22e1 HG_SOURCE=pull HG_URL=bundle:../full.hg 
pulling from bundle://../full.hg
requesting all changes
adding changesets
adding manifests
adding file changes
added 9 changesets with 7 changes to 4 files (+1 heads)
(run 'hg heads' to see heads, 'hg merge' to merge)
====== Create partial clones
requesting all changes
adding changesets
adding manifests
adding file changes
added 4 changesets with 4 changes to 1 files
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
====== Log -R full.hg in partial
changeset:   8:836ac62537ab
tag:         tip
parent:      3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3m

changeset:   7:80fe151401c2
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3m

changeset:   6:1e3f6b843bd6
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3

changeset:   5:024e4e7df376
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.2

changeset:   4:5f4f3ceb285e
parent:      0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.1

changeset:   3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3

changeset:   2:d62976ca1e50
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.2

changeset:   1:10b2180f755b
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.1

changeset:   0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.0

====== Incoming full.hg in partial
comparing with bundle://../full.hg
searching for changes
changeset:   4:5f4f3ceb285e
parent:      0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.1

changeset:   5:024e4e7df376
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.2

changeset:   6:1e3f6b843bd6
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3

changeset:   7:80fe151401c2
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3m

changeset:   8:836ac62537ab
tag:         tip
parent:      3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3m

====== Outgoing -R full.hg vs partial2 in partial
comparing with ../partial2
searching for changes
changeset:   4:5f4f3ceb285e
parent:      0:5649c9d34dd8
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.1

changeset:   5:024e4e7df376
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.2

changeset:   6:1e3f6b843bd6
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3

changeset:   7:80fe151401c2
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     1.3m

changeset:   8:836ac62537ab
tag:         tip
parent:      3:ac69c658229d
user:        test
date:        Mon Jan 12 13:46:40 1970 +0000
summary:     0.3m

====== Outgoing -R does-not-exist.hg vs partial2 in partial
abort: No such file or directory: ../does-not-exist.hg
====== Unbundle incremental bundles into fresh empty in one go
1 changesets found
1 changesets found
adding changesets
adding manifests
adding file changes
added 1 changesets with 1 changes to 1 files
adding changesets
adding manifests
adding file changes
added 1 changesets with 1 changes to 1 files
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
====== test for 540d1059c802
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
searching for changes
1 changesets found
comparing with ../bundle.hg
searching for changes
changeset:   2:ed1b79f46b9a
tag:         tip
parent:      0:bbd179dfa0a7
user:        test
date:        Thu Jan 01 00:00:00 1970 +0000
summary:     change foo

